x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/VerifierTests.java

```diff
@@ -1360,6 +1360,15 @@ public void testToDatePeriodToTimeDurationWithInvalidType() {
         );
     }
 
+    public void testNonMetadataScore() {
```

Comment:
I think this check in `EsPhysicalOperationProviders` should be sufficient to avoid population of _score:
```
...
if (EsqlCapabilities.Cap.METADATA_SCORE.isEnabled() && esQueryExec.scoring()) {
...
```
because `esQueryExec.scoring()` checks for presence of the MetadataAttribute but the explicit capability check should ensure usage in snapshot builds only.
However if we want to fail at logical plan, I can go with adding the check (but it needs to happen in `LogicalPlanBuider` since `esql-core` can't depend on `esql` module).